pymc/distributions/distribution.py

```diff
@@ -622,13 +679,16 @@ def dist(
         if logcdf is None:
             logcdf = default_not_implemented(class_name, "logcdf")
 
+        def dist_moment(rv, size, *dist_params):
+            # TODO: add check for other op with inner graph (not Scan)
+            fgraph = FunctionGraph(outputs=[dist(*dist_params, size=size)], clone=True)
```

Comment:
Should we specify the FunctionGraph inputs? I guess something like:

```
rv = dist(*dis_params, size=size) 
# Filter shared variables / constants that I think cannot be inputs to the FunctionGraph
filtered_inputs = [inp for inp in rv.owner.inputs if inp...]
fgraph = FunctionGraph(inputs=filtered_inputs, outputs=rv, clone=True)
```

Reply:
I experimented with this approach a little bit and it is not working. We need a whole graph, but `rv.owner.inputs` get us only last node and I got `MissingInputError`. For example for  `pt.exp(pm.Normal.dist(mu, sigma, size=size))` dist function `rv.owner.inputs` is just `[normal_rv{0, (0, 0), floatX, False}.out]`

Reply:
Hmm, that doesn't sound right. `dist` in a CustomDist should return an `OpFromGraph`, whose inputs would be `size` `mu`, `sigma`, and the shared RNGs.

```python
import pymc as pm

def dist_fn(mu, sigma, size):
    return pm.math.exp(pm.Normal.dist(mu, sigma, shape=size))

rv = pm.CustomDist.dist(0, 1, dist=dist_fn)
rv.owner.inputs
# [TensorConstant(TensorType(int64, shape=(0,)), data=array([], ... ype=int64)),
#  TensorConstant(TensorType(int8, shape=()), data=array(0, dtype=int8)),
#  TensorConstant(TensorType(int8, shape=()), data=array(1, dtype=int8)),
#  RandomGeneratorSharedVariable(<Generator(PCG64) at 0x7F61326D0820>)]
```

The point is that we don't want to rewrite the graph above the CustomDist.

Reply:
Actually we can just apply the moment in the rv.fgraph (after cloning it)?

Reply:
@ricardoV94  now we take output of `dist_fn` (not `CustomDist.dist`), build fgraph  and apply moment to it nodes. Do you think we need to take `CustomDist.dist` instead of `dist_fn`?

Reply:
Will add comment

Reply:
> @ricardoV94  now we take output of `dist_fn` (not `CustomDist.dist`), build fgraph  and apply moment to it nodes. Do you think we need to take `CustomDist.dist` instead of `dist_fn`? 

Yes that's better, because it already takes care of encapsulating the graph that corresponds to the CustomDist

Reply:
I think we can not use `CustomDist.dist` because we need to build `moment` inside `dist` method

Reply:
Can we extract the code that creates the fgraph from user provided fn and inputs into a helper outside of `CustomDist.dist`? That was the only reason I was suggesting we call it again.

Reply:
@ricardoV94 I found that `FunctionGraph` can extract and filter graph inputs on its own:
 ```
         if inputs is None:
            inputs = [
                i for i in graph_inputs(outputs) if not isinstance(i, AtomicVariable)
            ]
```
 https://github.com/pymc-devs/pytensor/blob/7bb18f3a3590d47132245b7868b3a4a6587a4667/pytensor/graph/fg.py#L105
 so I think we need not to specify inputs manually. But I moved graph construction in helper function anyway.